### PR TITLE
Proper linting

### PR DESCRIPTION
**Pull request by ppershing**

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+node_modules
+lib
+tests/lib
```
Comment (refi93) [inline .eslintignore:3]:
```suggestion
test/lib
example-node/lib
```

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  "plugins": ["flowtype", "import", "prettier", "mocha"],
+  "rules": {
+    "no-unused-vars": [
+      "warn",
+      {
+        "varsIgnorePattern": "^_",
+        "argsIgnorePattern": "^_"
+      }
+    ],
+    "flowtype/boolean-style": [2, "boolean"],
```
Comment (refi93) [inline .eslintrc:18]: a bit of a nitpick but would be perhaps nice to replace the numerical constants with its verbal equivalents - warn/error for the sake of clarity

File: workspace.code-workspace [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+	"folders": [
+		{
+			"path": "."
```
Comment (refi93) [inline workspace.code-workspace:4]: don't we want to exclude lib/node modules so it doesn't slow search down/pollute it with meaningless results?

File: .eslintignore (hunk 1 above)
Comment (refi93) [reply .eslintignore:3]: the original patterns, especially "lib" does not work probably as intended as it matches any file/folder with that name regardless of where in the file tree it actually is: https://www.atlassian.com/git/tutorials/saving-changes/gitignore

**Review by refi93 — CHANGES_REQUESTED** [review]
a few improvement suggestions and also `yarn lint` fails with two automatically fixable errors in `/test/src/integration/getExtendedPublicKey.js` - missing semicolon

The rest of the eslint/flow errors probably makes sense to tackle in a separate PR

File: .eslintrc (hunk 2 above)
Comment (ppershing) [reply .eslintrc:18]: fixed

File: workspace.code-workspace (hunk 3 above)
Comment (ppershing) [reply workspace.code-workspace:4]: I would say - if you have any workspace preferences, go for it as a followup PR.

**Comment by ppershing** [comment]
> a few improvement suggestions and also `yarn lint` fails with two automatically fixable errors in `/test/src/integration/getExtendedPublicKey.js` - two missing semicolons
> 
> The rest of the eslint/flow errors probably makes sense to tackle in a separate PR

For now I will be skipping tests (as long as they work). They will need a huge pass once we switch to TS anyway ...

**Review by refi93 — APPROVED** [review]

**Pull request merged**